src/random.rs

```diff
@@ -0,0 +1,33 @@
+use std::thread;
+use std::time::Duration;
+
+use getrandom::getrandom;
+
+pub fn fill_random(dest: &mut [u8]) {
+    let mut retried = false;
+    loop {
+        if Ok(()) == getrandom(dest) {
+            return;
+        }
+
+        if !retried {
```

Comment:
So, very new to rust, take this with a pinch of salt.
I see getrandom() does a libc call to getrandom(), which according to the manual by default uses urandom as source and guarantees to block if the pool has not been initialized, and to always return up to 256 bytes of data.
As we are requesting 16bytes if I read the code correctly, why is the retry needed? Won't the getramdon() call either block or succeed but not fail?

Reply:
@alexgg that was really just in case as I was not sure it may fail even though it says it will block.